contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.1.0--4.2.0.sql

```diff
@@ -38,6 +38,67 @@ LANGUAGE plpgsql;
  * final behaviour.
  */
 
+-- helper functions for upper and lower --
+CREATE OR REPLACE FUNCTION sys.upper_helper(sys.VARCHAR)
+RETURNS sys.VARCHAR
+AS 'babelfishpg_tsql', 'tsql_upper' LANGUAGE C IMMUTABLE PARALLEL SAFE;
+
+CREATE OR REPLACE FUNCTION sys.lower_helper(sys.VARCHAR)
+RETURNS sys.VARCHAR
+AS 'babelfishpg_tsql', 'tsql_lower' LANGUAGE C IMMUTABLE PARALLEL SAFE;
+
+-- upper --
+CREATE OR REPLACE FUNCTION sys.upper(ANYELEMENT)
+RETURNS sys.VARCHAR
+AS $$
+DECLARE
+    varch sys.varchar;
+BEGIN
+    IF pg_typeof($1) IN ('image'::regtype, 'sql_variant'::regtype, 'xml'::regtype, 'geometry'::regtype, 'geography'::regtype) THEN
+        RAISE EXCEPTION 'Argument data type % is invalid for argument 1 of upper function.', pg_typeof($1);
```

Comment:
Have we added test cases for all these data types

Reply:
Yes, we have already added tests for all these data types